### PR TITLE
fastnlo_toolkit: fix on darwin

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

ZHF #144627

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Comment by risicle** [comment]
(python packages still broken though)

File: pkgs/development/libraries/physics/fastnlo_toolkit/default.nix [hunk 1]
```diff
@@ -53,6 +38,10 @@ stdenv.mkDerivation rec {
   preConfigure = ''
     substituteInPlace ./fastnlotoolkit/Makefile.in \
       --replace "-fext-numeric-literals" ""
+
+    # disable test that fails due to strict floating-point number comparison
+    substituteInPlace ./check/Makefile.in \
+      --replace "fnlo-tk-stattest.pl" ""
```
Comment (Mic92) [inline pkgs/development/libraries/physics/fastnlo_toolkit/default.nix:44]: Might be more robust than patching Makefiles:

```suggestion
    echo "#!/usr/bin/env perl" > check/fnlo-tk-stattest.pl.in
    chmod +x check/fnlo-tk-stattest.pl.in
```

File: pkgs/development/libraries/physics/fastnlo_toolkit/default.nix [hunk 2]
```diff
@@ -14,28 +13,14 @@
...
-
-  postPatch = ''
-    # remove duplicate macro, to fix for autoconf 2.70
-    sed -e '0,/AC_CONFIG_MACRO_DIR\([m4]\)/{/AC_CONFIG_MACRO_DIR/d}' -i configure.ac
-  '';
+  version = "2.5.0-2826";
 
-  nativeBuildInputs = [ autoreconfHook ];
+  src = fetchurl {
+    url = "https://fastnlo.hepforge.org/code/v25/fastnlo_toolkit-2.5.0-2826.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/physics/fastnlo_toolkit/default.nix:21]: Please substitute the version.

File: pkgs/development/libraries/physics/fastnlo_toolkit/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/development/libraries/physics/fastnlo_toolkit/default.nix]: Sure, that will report test as if it ran, though

File: pkgs/development/libraries/physics/fastnlo_toolkit/default.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/development/libraries/physics/fastnlo_toolkit/default.nix]: Done

**Pull request merged**